    # Publish joint states
    joint_state_controller:
        type: joint_state_controller/JointStateController
        publish_rate: 100

    panda_arm_controller:
        type: position_controllers/JointTrajectoryController
        joints:
            - panda_joint1
            - panda_joint2
            - panda_joint3
            - panda_joint4
            - panda_joint5
            - panda_joint6
            - panda_joint7

        constraints:
            goal_time: 2.0

        state_publish_rate: 100

    panda_hand_controller:
        type: position_controllers/JointTrajectoryController
        joints:
            - panda_finger_joint1
            - panda_finger_joint2
        gains:
            panda_finger_joint1: { p: 100, d: 1, i: 1, i_clamp: 1 }
            panda_finger_joint2: { p: 100, d: 1, i: 1, i_clamp: 1 }
        constraints:
          stopped_velocity_tolerance: 0.015 # Override default
          goal_time: 2.0
          panda_finger_joint1: {goal: 1 , trajectory: 1}
          panda_finger_joint2: {goal: 1 , trajectory: 1}
        stop_trajectory_duration: 0.5
        state_publish_rate:  125
        action_monitor_rate: 10
        velocity_ff:
          panda_finger_joint1: 10.0
          panda_finger_joint2: 10.0

    joint_position_controller:
        type: panda_simulation/JointPositionController
        arm_id: panda
        joint_names:
            - panda_joint1
            - panda_joint2
            - panda_joint3
            - panda_joint4
            - panda_joint5
            - panda_joint6
            - panda_joint7
        gains: [
            1, 1, 1, 1, 1, 1, 1
        ]
